test.py

```diff
@@ -78,6 +78,15 @@ def nocolor(text: str) -> str:
         return palette.ansi_escape.sub('', text)
 
 
+class TestCase(NamedTuple):
+    suite_name: str
+    name: str
```

Comment:
And here the ambiguity from the previous rename is striking us. Suite_name is fine, name should be "file_name", and "case" should be "case_name". Or, if you like brevity, suite, file, case.

Reply:
OK, `file_name` and `case_name`.

Reply:
So @alecco the commit message contains a new term, "test instance", you also use "TestDef" elsewhere in our discussion, is "Def" what "defines" a test instance?

Reply:
Perhaps "Test instance" is a "TestTask", and "TestDef" is a "TestTaskDef". In a comment for TestTask we could clarify that a TestTask can stand for an entire test file or a single test case. Alternatively we could say that we always run an entire test file and test case selection is a filter that we push down to the runner to filter out unnecessary cases. In this case the terminology would be TestFileTask and TestFileTaskDef or something along these lines. WDYT?

Reply:
The (current) docstring for `class TestCase`'s `.match()` method compares 2 object instances of TestCase. The this/self instance and the other one passed in the parameters. This is Python terminology.

Reply:
> Perhaps "Test instance" is a "TestTask", and "TestDef" is a "TestTaskDef". In a comment for TestTask we could clarify that a TestTask can stand for an entire test file or a single test case. Alternatively we could say that we always run an entire test file and test case selection is a filter that we push down to the runner to filter out unnecessary cases. In this case the terminology would be TestFileTask and TestFileTaskDef or something along these lines. WDYT?

TestFileTask` is OK. What could be the name for the pattern matching class? (built from the command line "name" specification).